cli/pkg/helm/client.go

```diff
@@ -0,0 +1,557 @@
+		jsonCRD, err := yaml.ToJSON(crd.File.Data)
+		if err != nil {
+			return err
+		}
+
+		var meta metaV1.TypeMeta
+		err = json.Unmarshal(jsonCRD, &meta)
+		if err != nil {
+			return err
+		}
+
+		switch meta.APIVersion {
+
+		case "apiextensions.k8s.io/apiextensionsV1":
+			var crdObj apiextensionsV1.CustomResourceDefinition
+			err = json.Unmarshal(jsonCRD, &crdObj)
+			if err != nil {
+				return err
+			}
+			existingCRDObj, err := k8sClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, crdObj.Name, metaV1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			crdObj.ResourceVersion = existingCRDObj.ResourceVersion
+			_, err = k8sClient.ApiextensionsV1().CustomResourceDefinitions().Update(ctx, &crdObj, metaV1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+
+		case "apiextensions.k8s.io/v1beta1":
+			var crdObj v1beta1.CustomResourceDefinition
+			err = json.Unmarshal(jsonCRD, &crdObj)
+			if err != nil {
+				return err
+			}
+			existingCRDObj, err := k8sClient.ApiextensionsV1().CustomResourceDefinitions().Get(ctx, crdObj.Name, metaV1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			crdObj.ResourceVersion = existingCRDObj.ResourceVersion
+			_, err = k8sClient.ApiextensionsV1beta1().CustomResourceDefinitions().Update(ctx, &crdObj, metaV1.UpdateOptions{})
+			if err != nil {
+				return err
+			}
+
+		default:
+			return fmt.Errorf("failed to update crd %q: unsupported api-version %q", crd.Name, meta.APIVersion)
+		}
+	}
+
+	return nil
+}
+
+// getChart returns a chart matching the provided chart name and options
+func (c *HelmClient) getChart(chartName string, chartPathOptions *action.ChartPathOptions) (*chart.Chart, string, error) {
+	chartPath, err := chartPathOptions.LocateChart(chartName, c.Settings)
+	if err != nil {
+		return nil, "", err
+	}
+
+	helmChart, err := loader.Load(chartPath)
+	if err != nil {
+		return nil, "", err
+	}
+
+	if helmChart.Metadata.Deprecated {
+		log.Printf("WARNING: This chart (%q) is deprecated", helmChart.Metadata.Name)
+	}
+
+	return helmChart, chartPath, err
+}
+
+// chartIsInstalled checks whether a chart is already installed or not by the provided release name
+func (c *HelmClient) chartIsInstalled(release string) (bool, error) {
+	histClient := action.NewHistory(c.ActionConfig)
+	histClient.Max = 1
+	if _, err := histClient.Run(release); err == driver.ErrReleaseNotFound {
+		return false, nil
+	} else if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (c *HelmClient) listDeployedReleases() ([]*release.Release, error) {
+	err := c.ActionConfig.Init(c.Settings.RESTClientGetter(), "", "", func(_ string, _ ...interface{}) {})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list deployed releases %v", err)
+	}
+	listClient := action.NewList(c.ActionConfig)
+	listClient.StateMask = action.ListDeployed
+	return listClient.Run()
+}
+
+func (c *HelmClient) getReleaseValues(name string, allValues bool) (map[string]interface{}, error) {
```

Comment:
Since I don't see any case that you are not getting all values, can we remove this bool flag? Its an anti-pattern to send flags like this. If there is a need to have both versions (all values and not all values), I suggest creating two functions: `getAllReleaseValues` and `getReleaseValues`.

Makes the code easier to read.